File: backend/src/main/java/szakdoga/backend/app/module/Basket.java
package szakdoga.backend.app.module;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "Basket")
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Data
public class Basket extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String foodname;

    @Column(nullable = false)
    private String foodingredients;

    @Column(nullable = false)
    private int price;
}

File: backend/etterem-frontend/src/app/entity/Basket.ts
export class Basket {

  public id:number;
  public name:String;
  public foodname:String;
  public foodingredients:String;
  public price: number;


  constructor(id?:number,name?: String, foodname?: String, foodingredients?: String, price?: number) {
    this.id=id;
    this.name = name || "";
    this.foodname = foodname || "";
    this.foodingredients = foodingredients || "";
    this.price = price || 0;
  }
}
